File: html/old/backup/sacor2_2008/checksums.java
// SACO Second Round 2008
// Problem: Checksums
// Author:  Marco Gallotta

import java.io.*;
import java.util.Scanner;
public class checksums {
    public static void main(String args[]) {
	Scanner in = new Scanner(System.in);
	System.out.print("Enter credit card number: ");
	String cc = in.next();
	int sum = 0;
	for (int i = 0; i < cc.length(); i += 2)
	    sum += cc.charAt(i) - '0';
	for (int i = 1; i < cc.length(); i += 2) {
	    int dbl = (cc.charAt(i) - '0') * 2;
	    sum += dbl / 10 + dbl % 10;
	}
	
	if(sum % 10 == 0)
	    System.out.println(sum + " VALID");
	else
	    System.out.println(sum + " INVALID");
    }
}

File: html/old/backup/sacor2_2008/checksums.cpp
// SACO 2nd Round 2008
// Checksums Solution by Keegan Carruthers-Smith

#include <iostream>
#include <string>

using namespace std;

int main() {
    string number;
    int checksum = 0;

    cout << "Enter credit card number: ";
    cin >> number;

    for (int i = 0; i < number.size(); i++) {
	if (i % 2 == 1) {
	    int k = (number[i] - '0') * 2;
	    checksum += (k % 10);
	    checksum += (k / 10);
	} else {
	    checksum += (number[i] - '0');
	}
    }

    if (checksum % 10 == 0)
	cout << checksum << " VALID" << endl;
    else
	cout << checksum << " INVALID" << endl;

    return 0;
}
